src/App.svelte

```diff
@@ -0,0 +1,115 @@
+<script>
+	let display = 0;
+	let firstOperand = null;
+	let secondOperand = false;
+	let operator = null;
+	let operatorClicked = false;
+
+	const addDisplay = (num) => {
+		if(display === 0 || operatorClicked ) {
+			display = '';	
+		}
+		if(display.length <= 13){
+			display += String(num);	
+		}
+		operatorClicked = false;
+	}
+
+	const dot = () => {
+		if(display == 0) {
+			display = '0.';
+		}else if(!display.includes('.')){
+			display += '.';
+		}	
+	}
+
+	const resetDisplay = () => {
+		display = 0;
+	}
+
+	const deleteNum = () => {
+		display = String(display);
+		if(display.length <= 1) { 
+			display = 0;
+		}else if (display == /.$/) {
+			display = display.slice(0, -2); 
+		}else{
+			display = display.slice(0, -1);
+		}	
+	}	
+	
+	const equal = () => {
+		//debugger;
+		secondOperand = Number(display)
+		if(operator === '+'){
+		 	display = firstOperand + secondOperand;
+			display = display.toFixed(7).replace(/0+$/, '');
+		}else if(operator === '-'){
+		 	display =  firstOperand - secondOperand;
+			display = display.toFixed(7).replace(/0+$/, '');
+		}else if(operator === '*'){
+		 	display =  firstOperand * secondOperand;
+			display = display.toFixed(7).replace(/0+$/, '');
+		}else if(operator === '/'){
+		 	display =  firstOperand / secondOperand;
+			display = display.toFixed(7).replace(/0+$/, '');
+		}
+	} 
+
+	const operationsState = (operation) => {
+		if(secondOperand, operatorClicked = true){
```

Comment:
This seems very suspicious. `operatorClicked = true` would always return true. You are not comparing with `==` or `===` there. You are assigning with `=`. Also what's the point of secondOperand before comma there?
Did you mean if both secondOperand exists and operatorClicked equals to true? If so you should use `&&` operator between them.

Reply:
I wanted to set true for both by putting a comma after the secondOperator. I couldn't run the function correctly using the && operator. So I moved operatorClicked up and rearranged the function by defining secondOperator as Number.